File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Foundation/TimeoutBlockingQueueTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4oUnit;
using Db4objects.Db4o.Foundation;
using Db4objects.Db4o.Tests.Common.Foundation;

namespace Db4objects.Db4o.Tests.Common.Foundation
{
	public class TimeoutBlockingQueueTestCase : Queue4TestCaseBase
	{
		public virtual void TestTimeoutNext()
		{
			ITimeoutBlockingQueue4 queue = new TimeoutBlockingQueue(300);
			queue.Pause();
			queue.Add(new object());
			queue.Check();
			Assert.IsNull(queue.TryNext());
			Runtime4.SleepThrowsOnInterrupt(500);
			queue.Check();
			Assert.IsNotNull(queue.TryNext());
		}
	}
}

File: db4oj.tests/src/com/db4o/db4ounit/common/foundation/TimeoutBlockingQueueTestCase.java
/* Copyright (C) 2010 Versant Inc. http://www.db4o.com */
package com.db4o.db4ounit.common.foundation;

import com.db4o.foundation.*;

import db4ounit.*;

public class TimeoutBlockingQueueTestCase extends Queue4TestCaseBase {
	
	public void testTimeoutNext() {
		
		TimeoutBlockingQueue4<Object> queue = new TimeoutBlockingQueue<Object>(300);
		
		queue.pause();
		
		queue.add(new Object());
		
		queue.check();
		
		Assert.isNull(queue.tryNext());
		
		Runtime4.sleepThrowsOnInterrupt(500);
		queue.check();
		
		Assert.isNotNull(queue.tryNext());
		
	}
	
}
